index.d.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import Store from '@orbit/store';
+import { TransformOrOperations, QueryBuilder, Source } from '@orbit/data';
+
+
+export interface DataProviderProps {
+  dataStore: Store;
+}
+
+export interface RecordsToProps {
+  [key: string]: (q: QueryBuilder) => any;
+}
+
+export interface WithData {
+  dataStore: Store;
+  sources: { [sourceName: string]: Source };
+}
+
+export type WithDataProps =
+  & {
+    queryStore: (transformOrOperations: TransformOrOperations, options?: object, id?: string) => any;
+    updateStore: (transformOrOperations: TransformOrOperations, options?: object, id?: string) => any;
+    dataStore: Store;
+    sources: { [sourceName: string]: Source };
+  }
+  & WithData
```

Comment:
I'm pretty new to typescript (in fact never used it myself), but why is this intersection needed? Isn't it already present in the definition preceding this?